xbmc/addons/AddonDll.cpp

```diff
@@ -236,6 +240,62 @@ ADDON_STATUS CAddonDll::Create(ADDON_TYPE type, void* funcTable, void* info)
   return status;
 }
 
+ADDON_STATUS CAddonDll::Create(KODI_HANDLE firstKodiInstance)
+{
+  CLog::Log(LOGDEBUG, "ADDON: Dll Initializing - %s", Name().c_str());
+  m_initialized = false;
+
+  if (!LoadDll())
+  {
+    return ADDON_STATUS_PERMANENT_FAILURE;
+  }
+
+  /* Check versions about global parts on add-on (parts used on all types) */
+  for (unsigned int id = ADDON_GLOBAL_MAIN; id <= ADDON_GLOBAL_MAX; ++id)
+  {
+    if (!CheckAPIVersion(id))
+      return ADDON_STATUS_PERMANENT_FAILURE;
+  }
+
+  /* Allocate the helper function class to allow crosstalk over
+     helper add-on headers */
+  if (!InitInterface(firstKodiInstance))
+    return ADDON_STATUS_PERMANENT_FAILURE;
+
+  /* Call Create to make connections, initializing data or whatever is
+     needed to become the AddOn running */
+  ADDON_STATUS status = m_pDll->Create(&m_interface, nullptr);
+  if (status == ADDON_STATUS_OK)
+  {
+    m_initialized = true;
+  }
+  else if ((status == ADDON_STATUS_NEED_SETTINGS) || (status == ADDON_STATUS_NEED_SAVEDSETTINGS))
+  {
+    m_needsavedsettings = (status == ADDON_STATUS_NEED_SAVEDSETTINGS);
+    if ((status = TransferSettings()) == ADDON_STATUS_OK)
+      m_initialized = true;
+    else
+      new CAddonStatusHandler(ID(), status, "", false);
```

Comment:
The next part who must be removed asap, also a remnant from the old system who comes with c&p to there.

If acceptable for you becomes this done on soon coming requests. Is related to my comment before (only be used for this "ADDON_STATUS_NEED_SAVEDSETTINGS" and "ADDON_STATUS_NEED_SETTINGS")